Clerk Elements (Beta) - Initial docs (#890)
clerk/clerk-docs: docs/elements/examples/sign-in.mdx

---
description: Examples of pre-built sign-in flows with Clerk Elements.
---

> # Sign-in

Reviewer: ```suggestion
# Clerk Elements sign-in examples
```

> ## Basic sign-in

Reviewer: ```suggestion
```
When we add more than one example we can re-add this H2

                />
              </svg>
              Login with Google
            </Clerk.Connection>
          </div>
          <p className="text-center text-sm text-neutral-400">
            No account?{' '}
            <a
              href="#"
              className="font-medium text-white decoration-white/20 underline-offset-4 outline-none hover:underline focus-visible:underline"
            >
              Create an account
            </a>
          </p>
        </SignIn.Step>
      </SignIn.Root>
    </div>
  );
}
```
</CodeBlockDemo>

## Multi-factor authentication (MFA)

The following example demonstrates a simple multi-factor authentication (MFA) sign-in flow. The user has to provide their email and password and then verify their sign-in attempt with a phone code.

Before you build your sign-in flow, you need to configure the appropriate settings in Clerk:

1. Navigate to the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/email-phone-username).
1. In the navigation sidebar, select **User & Authentication > Email, Phone, Username**.

> 1. Ensure that **Email address**, **Password**, and **Phone number** are enabled.

Reviewer 2: ```suggestion
1. Ensure that **Email address**, **Phone number**, and **Password** are enabled.
```

> 1. Ensure that **Email address**, **Phone number**, and **Password** are enabled.

Reviewer 2: ```suggestion
1. Ensure that **Email address**, **Phone number**, and **Password** are enabled.
1. Next to **Phone number**, select the settings icon and enable **Require**.
```

> 1. In the **Contact information** section, ensure that **Email address**, **Password**, and **Phone number** are enabled.

Reviewer 2: ```suggestion
1. Ensure that **Email address**, **Password**, and **Phone number** are enabled.
```

            >
              <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 17 16" className="w-4" aria-hidden>
                <path
                  fill="currentColor"
                  d="M8.82 7.28v2.187h5.227c-.16 1.226-.57 2.124-1.192 2.755-.764.765-1.955 1.6-4.035 1.6-3.218 0-5.733-2.595-5.733-5.813 0-3.218 2.515-5.814 5.733-5.814 1.733 0 3.005.685 3.938 1.565l1.538-1.538C12.998.96 11.256 0 8.82 0 4.41 0 .705 3.591.705 8s3.706 8 8.115 8c2.382 0 4.178-.782 5.582-2.24 1.44-1.44 1.893-3.475 1.893-5.111 0-.507-.035-.978-.115-1.369H8.82Z"
                />
              </svg>
              Login with Google
            </Clerk.Connection>
          </div>
          <p className="text-center text-sm text-neutral-400">
            No account?{' '}
            <a
              href="#"
              className="font-medium text-white decoration-white/20 underline-offset-4 outline-none hover:underline focus-visible:underline"
            >
              Create an account
            </a>
          </p>
        </SignIn.Step>
      </SignIn.Root>
    </div>
  );
}
```
</CodeBlockDemo>

## Multi-factor authentication (MFA)

> The following example demonstrates a simple multi-factor authentication (MFA) sign-in flow. The user has to provide their email and password and then verify their sign-in attempt with a phone code.

Reviewer 2: ```suggestion
The following example demonstrates a simple multi-factor authentication (MFA) sign-in flow. The user can sign in with their email and password. If they have two-factor authentication enabled, they will need to verify their sign-in attempt with an SMS code.
```
